File: gmall/gmall-realtime/src/main/java/org/pcchen/utils/PropertiesUtils.scala
package org.pcchen.utils

import java.io.InputStreamReader
import java.util.Properties

/**
  *
  * @author ceek
  * @create 2021-03-02 14:39
  **/
object PropertiesUtils {
  def main(args: Array[String]): Unit = {
    val properties: Properties = PropertiesUtils.load("config.properties")

    println(properties.getProperty("kafka.broker.list"))
  }

  def load(propertieName: String): Properties = {
    val prop = new Properties();
    prop.load(new InputStreamReader(Thread.currentThread().getContextClassLoader.getResourceAsStream(propertieName), "UTF-8"))
    prop
  }
}

File: gmall/gmall-common/src/main/java/org/pcchen/constants/PropertiesUtils.java
package org.pcchen.constants;

import java.io.IOException;
import java.util.Properties;

/**
 * @author ceek
 * @date 2021/3/3 20:41
 */
public class PropertiesUtils {
    public static String getProperties(String params) {
        Properties properties = new Properties();
        try {
            properties.load(Thread.currentThread().getContextClassLoader().getResourceAsStream("config.properties"));
        } catch (IOException e) {
            e.printStackTrace();
        } finally {
            return properties.getProperty(params);
        }
    }

    public static void main(String[] args) {
        System.out.println(PropertiesUtils.getProperties("es.host"));
    }
}